packages/react/src/components/Animation/Animation.tsx

```diff
@@ -102,23 +127,65 @@ class Animation extends UIComponent<WithAsProp<AnimationProps>, any> {
     keyframeParams: PropTypes.object,
     playState: PropTypes.string,
     timingFunction: PropTypes.string,
+    visible: PropTypes.bool,
+    mountOnEnter: PropTypes.bool,
+    appear: PropTypes.bool,
+    unmountOnExit: PropTypes.bool,
+    timeout: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.shape({
+        appear: PropTypes.number,
+        enter: PropTypes.number,
+        exit: PropTypes.number,
+      }),
+    ]),
+  }
+
+  static defaultProps = {
+    timeout: 0,
   }
 
   renderComponent({ ElementType, classes, unhandledProps }) {
-    const { children } = this.props
+    const {
+      children,
+      mountOnEnter,
+      unmountOnExit,
+      timeout,
+      appear,
+      visible,
+      onEnter,
+      onEntering,
+      onEntered,
+      onExit,
+      onExited,
+      onExiting,
+    } = this.props
 
     const child =
       childrenExist(children) && (React.Children.only(children) as React.ReactElement<any>)
     const result = child
       ? React.cloneElement(child, {
-          className: cx(child.props.className, classes.children),
+          ...unhandledProps,
+          className: cx(classes.root, child.props.className),
         })
       : ''
 
     return (
-      <ElementType className={classes.root} {...unhandledProps}>
-        {result}
-      </ElementType>
+      <Transition
+        in={visible}
+        appear={appear}
+        mountOnEnter={mountOnEnter}
+        unmountOnExit={unmountOnExit}
+        timeout={timeout}
+        onEnter={onEnter}
+        onEntering={onEntering}
+        onEntered={onEntered}
+        onExit={onExit}
+        onExiting={onExiting}
+        onExited={onExited}
+      >
+        {state => result}
```

Comment:
I am currently not using the state, as in the usage I am changing the animation. We don't have the children as function concept anywhere else so far, so I didn't add it for the Animation component. (it would be easy to migrate if we decide in the future!)

Reply:
Does this work for adding/removing table items within the Table component or tree items within Tree? Or do they need to be state aware?

Reply:
It should work, if not, we can Implement the AnimationGroup component in a similar way of how the TransitionGroup (https://reactcommunity.org/react-transition-group/transition-group) component is implemented from the same library). But, to be honest, I would try to avoid adding different APIs and try to implement it with this API